src/Parsers/ASTCreateQuery.cpp

```diff
@@ -329,127 +334,133 @@ void ASTCreateQuery::formatQueryImpl(const FormatSettings & settings, FormatStat
-        settings.ostr << (settings.hilite ? hilite_keyword : "") << action << " DICTIONARY "
-                      << (if_not_exists ? "IF NOT EXISTS " : "") << (settings.hilite ? hilite_none : "")
-                      << (database ? backQuoteIfNeed(getDatabase()) + "." : "") << backQuoteIfNeed(getTable());
+        out.writeKeyword(action);
+        out.writeKeyword(" DICTIONARY ");
+        out.writeKeyword(if_not_exists ? "IF NOT EXISTS " : "");
+        out.ostr << (database ? backQuoteIfNeed(getDatabase()) + "." : "") << backQuoteIfNeed(getTable());
         if (uuid != UUIDHelpers::Nil)
-            settings.ostr << (settings.hilite ? hilite_keyword : "") << " UUID " << (settings.hilite ? hilite_none : "")
-                          << quoteString(toString(uuid));
-        formatOnCluster(settings);
+        {
+            out.writeKeyword(" UUID ");
+            out.ostr << quoteString(toString(uuid));
+        }
+        formatOnCluster(out);
     }
 
     if (to_table_id)
     {
         assert((is_materialized_view || is_window_view) && to_inner_uuid == UUIDHelpers::Nil);
-        settings.ostr
-            << (settings.hilite ? hilite_keyword : "") << " TO " << (settings.hilite ? hilite_none : "")
+        out.writeKeyword(" TO ");
+        out.ostr
             << (!to_table_id.database_name.empty() ? backQuoteIfNeed(to_table_id.database_name) + "." : "")
             << backQuoteIfNeed(to_table_id.table_name);
     }
 
     if (to_inner_uuid != UUIDHelpers::Nil)
     {
         assert(is_materialized_view && !to_table_id);
-        settings.ostr << (settings.hilite ? hilite_keyword : "") << " TO INNER UUID " << (settings.hilite ? hilite_none : "")
-                      << quoteString(toString(to_inner_uuid));
+        out.writeKeyword(" TO INNER UUID ");
+        out.ostr << quoteString(toString(to_inner_uuid));
     }
 
     if (!as_table.empty())
     {
-        settings.ostr
-            << (settings.hilite ? hilite_keyword : "") << " AS " << (settings.hilite ? hilite_none : "")
+        out.writeKeyword(" AS ");
+        out.ostr
             << (!as_database.empty() ? backQuoteIfNeed(as_database) + "." : "") << backQuoteIfNeed(as_table);
     }
 
     if (as_table_function)
     {
         if (columns_list && !columns_list->empty())
         {
-            frame.expression_list_always_start_on_new_line = true;
-            settings.ostr << (settings.one_line ? " (" : "\n(");
-            FormatStateStacked frame_nested = frame;
-            columns_list->formatImpl(settings, state, frame_nested);
-            settings.ostr << (settings.one_line ? ")" : "\n)");
-            frame.expression_list_always_start_on_new_line = false;
+            out.setExpressionListAlwaysStartsOnNewLine();
+            out.nlOrWs();
+            out.ostr << "(";
+            columns_list->formatImpl(out);
+            out.nlOrNothing();
+            out.ostr << ")";
+            out.setExpressionListAlwaysStartsOnNewLine(false);
```

Comment:
If you copy `out` at the beginning of the scope then you won't need to restore the original value of `ExpressionListAlwaysStartsOnNewLine` at the end of the scope.